### PR TITLE
Make /app good for for 0.74

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

This PR is a part of ongoing effort to bring Reanimated to React Native 0.74 with bridgeless mode. It consists of following Pull Requests, which are split for review convenience - but shouldn't be considered separate entities. They have to be merged in the following order:

- #5830
- #5815
- #5834
- #5734 

## Test plan

:shipit:

### CODE REVIEW

File: app/src/examples/AboutExample.tsx [hunk 1]
```diff
@@ -68,6 +68,10 @@ export default function AboutExample() {
             <Text style={styles.bold}>RN version:</Text>{' '}
             {getReactNativeVersion()}
           </Text>
+          <Text style={styles.text}>
+            <Text style={styles.bold}>Bridgeless enabled:</Text>{' '}
+            {(global as any)._IS_BRIDGELESS ? 'yes' : 'no'}
```
[the remark this answers is not in the document]
Comment (tjzel) [reply app/src/examples/AboutExample.tsx:73]: I did to print it in `AboutExample` 🤣. We can remove it if we don't want it.

[the remark this answers is not in the document]
Comment (piaskowyk) [reply app/src/examples/AboutExample.tsx:73]: I'm ok with that 😅

**Review by piaskowyk — APPROVED** [review]
For me is legit 🫡  Before merging:
- Let's sort out the CI - https://github.com/software-mansion/react-native-reanimated/actions/runs/8421856626/job/23059739159?pr=5830
- Wait for approval on https://github.com/software-mansion/react-native-reanimated/pull/5815, and then let's merge both pull requests together to keep the main branch in a consistent state.

File: app/package.json [hunk 2]
```diff
@@ -12,23 +12,22 @@
...
+    "react-native-gesture-handler": "^2.16.0-rc.0",
+    "react-native-pager-view": "^6.2.3",
     "react-native-reanimated": "link:../",
-    "react-native-safe-area-context": "^4.7.2",
-    "react-native-screens": "^3.27.0",
-    "react-native-svg": "^13.14.0",
+    "react-native-safe-area-context": "4.10.0-rc.0",
+    "react-native-screens": "git+https://github.com/software-mansion/react-native-screens.git#ddf267a4ce157dc60fa0bb94e8096c4d99f7e878",
+    "react-native-svg": "^15.2.0-rc.0",
     "react-native-web": "~0.18.12"
```
[the remark this answers is not in the document]
Comment (tjzel) [reply app/package.json]: I have them all upgraded in other PRs 👍

File: app/src/examples/LiquidSwipe/LiquidSwipe.tsx [hunk 3]
```diff
@@ -34,26 +34,21 @@ export default function LiquidSwipe() {
   const isBack = useSharedValue(false);
   const centerY = useSharedValue(initialWaveCenter);
   const progress = useSharedValue(0);
+  const ctxDragX = useSharedValue(0);
+  const ctxStartY = useSharedValue(0);
```
Comment (tomekzaw) [inline app/src/examples/LiquidSwipe/LiquidSwipe.tsx:38]: remove `ctx`

File: app/src/examples/AboutExample.tsx (hunk 1 above)
Comment (tomekzaw) [inline app/src/examples/AboutExample.tsx:73]: move to `isBridgeless()`

File: app/src/examples/ScreenStackHeaderConfigBackgroundColorExample.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ import React from 'react';
 
 const AnimatedScreenStackHeaderConfig = Animated.createAnimatedComponent(
   Platform.OS === 'web'
-    ? React.forwardRef(ScreenStackHeaderConfig)
+    ? React.forwardRef(ScreenStackHeaderConfig as any)
```
Comment (tomekzaw) [inline app/src/examples/ScreenStackHeaderConfigBackgroundColorExample.tsx:24]: 😢

**Review by tomekzaw — CHANGES_REQUESTED** [review]

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**